front/app/containers/Admin/reporting/components/ReportBuilder/TopBar/index.tsx

```diff
@@ -130,39 +127,42 @@ const ContentBuilderTopBar = ({
       return;
     }
 
-    const nodes = query.getSerializedNodes();
-    const firstNode = nodes.ROOT?.nodes[0];
+    const interval = setInterval(() => {
+      const nodes = query.getSerializedNodes();
+      const firstNode = nodes.ROOT?.nodes[0];
+      if (!firstNode) return;
 
-    if (!firstNode) return;
+      const displayName = nodes?.[firstNode].displayName;
 
-    const displayName = nodes?.[firstNode].displayName;
+      if (!['ProjectTemplate', 'PhaseTemplate'].includes(displayName)) {
+        // In theory this should not be possible, but handling
+        // it gracefully just in case
+        setInitialized(true);
+        clearInterval(interval);
+        return;
+      }
 
-    if (['ProjectTemplate', 'PhaseTemplate'].includes(displayName)) {
       const numberOfNodes = Object.keys(nodes).length;
-
       if (displayName === 'ProjectTemplate' && numberOfNodes < 5) return;
```

Comment:
Not related to the changes, but why don't we save in this case?

Reply:
This means that the template is partly loaded, but not fully yet. If it's fully ready it should have much more than 5 nodes. I remember this was an issue back when I added this code. Just a little bit of extra certainty that it's ready to be saved.